src/librbd/io/ImageRequestWQ.cc

```diff
@@ -466,6 +480,11 @@ void *ImageRequestWQ<I>::_void_dequeue() {
   {
     RWLock::RLocker locker(m_lock);
     bool write_op = peek_item->is_write_op();
```

Comment:
the throttle `if` conditional needs the value of `write_op` to check whether current op is write op or read op.   
So I don't think it can be moved after the `if` conditional.